src/commands/account/get_public_key/from_legacy_keychain/mod.rs

```diff
@@ -0,0 +1,133 @@
+use color_eyre::eyre::WrapErr;
+
+use crate::common::JsonRpcClientExt;
+use crate::common::RpcQueryResponseExt;
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = crate::GlobalContext)]
+#[interactive_clap(output_context = PublicKeyFromLegacyKeychainContext)]
+pub struct PublicKeyFromKeychain {
+    #[interactive_clap(skip_default_input_arg)]
+    /// For which account do you need to view the public key?
+    owner_account_id: crate::types::account_id::AccountId,
+    #[interactive_clap(named_arg)]
+    /// Select network
+    network_config: crate::network::Network,
+}
+
+#[derive(Clone)]
+pub struct PublicKeyFromLegacyKeychainContext(crate::network::NetworkContext);
+
+impl PublicKeyFromLegacyKeychainContext {
+    pub fn from_previous_context(
+        previous_context: crate::GlobalContext,
+        scope: &<PublicKeyFromKeychain as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let config = previous_context.config.clone();
+        let account_id = scope.owner_account_id.clone();
+
+        let on_after_getting_network_callback: crate::network::OnAfterGettingNetworkCallback =
+            std::sync::Arc::new({
+                move |network_config| {
+                    let keychain_folder = config
+                        .credentials_home_dir
+                        .join(&network_config.network_name);
+                    let signer_keychain_folder = keychain_folder.join(account_id.to_string());
+                    let signer_access_key_file_path: std::path::PathBuf = {
+                        if previous_context.offline {
+                            eprintln!(
+                            "\nThe signer's public key cannot be verified and retrieved offline."
```

Comment:
But why ? I still think we can easily process the directory and return user a key. Just with a warning that this key wasn't checked that it's active on chain

Reply:
We can easily process the directory and return the key to the user. But we promised the user that they would be able to get the public key with full access.
For example, an account has 10 key files. We give the user the public key and say that we don't know if it is active or has full access.